e2e/testcases/reconciler_finalizer_test.go

```diff
@@ -522,8 +522,26 @@ func TestReconcileFinalizerReconcileTimeout(t *testing.T) {
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
+	nt.T.Cleanup(func() {
+		namespace := &corev1.Namespace{}
+		if err := nt.KubeClient.Get(namespaceNN.Name, namespaceNN.Namespace, namespace); err != nil {
+			if apierrors.IsNotFound(err) { // Happy path - exit
+				return
+			}
+			nt.T.Error(err) // unexpected error
```

Comment:
this is fine, but fwiw, its safe to use Fatal in Cleanup. The caller catches panics and makes sure other cleanups get called, plus then you don't need the return.